tests/helpers/zos_lineinfile_helper.py

```diff
@@ -159,3 +186,126 @@ def DsGeneralResultKeyMatchesRegex(test_name, ansible_zos_module, test_env, test
         for key in kwargs:
             assert re.match(kwargs.get(key), result.get(key))
     clean_ds_test_env(test_env["DS_NAME"], hosts)
+
+
+def DsGeneralForce(ansible_zos_module, test_env, test_text, test_info, expected):
+    MEMBER_1, MEMBER_2 = "MEM1", "MEM2"
+    hosts = ansible_zos_module
+    test_info["path"] = DEFAULT_DATA_SET_NAME+"({0})".format(MEMBER_2)
+    if test_env["ENCODING"]:
+        test_info["encoding"] = test_env["ENCODING"]
+    try:
+        # set up:
+        # create pdse
+        hosts.all.zos_data_set(name=DEFAULT_DATA_SET_NAME, state="present", type="pdse", replace=True)
+        # add members
+        hosts.all.zos_data_set(
+            batch=[
+                {
+                    "name": DEFAULT_DATA_SET_NAME + "({0})".format(MEMBER_1),
+                    "type": "member",
+                    "state": "present",
+                    "replace": True,
+                },
+                {
+                    "name": DEFAULT_DATA_SET_NAME + "({0})".format(MEMBER_2),
+                    "type": "member",
+                    "state": "present",
+                    "replace": True,
+                },
+            ]
+        )
+        # write memeber to verify cases
+        hosts.all.shell(cmd="echo \"{0}\" > {1}".format(test_text, test_info["path"]))
+        # copy/compile c program and copy jcl to hold data set lock for n seconds in background(&)
+        hosts.all.zos_copy(content=c_pgm, dest='/tmp/disp_shr/pdse-lock.c', force=True)
+        hosts.all.zos_copy(
+            content=call_c_jcl.format(DEFAULT_DATA_SET_NAME, MEMBER_1),
+            dest='/tmp/disp_shr/call_c_pgm.jcl',
+            force=True
+        )
+        hosts.all.shell(cmd="xlc -o pdse-lock pdse-lock.c", chdir="/tmp/disp_shr/")
+
+        # submit jcl
+        hosts.all.shell(cmd="submit call_c_pgm.jcl", chdir="/tmp/disp_shr/")
+
+        # pause to ensure c code acquires lock
+        time.sleep(5)  
+        # call line infile to see results      
+        results = hosts.all.zos_lineinfile(**test_info)
+        pprint(vars(results))
+        if test_env["ENCODING"] == 'IBM-1047':
+            cmdStr =r"""cat "//'{0}'" """.format(test_info["path"])
+            results = hosts.all.shell(cmd=cmdStr)
+            pprint(vars(results))
+            for result in results.contacted.values():
+                print()
+                assert result.get("stdout").replace('\n', '').replace(' ', '') == expected.replace('\n', '').replace(' ', '')
```

Comment:
also noticed no check of `results['changed']` unlike the other ones -- was this an intentional test design choice?

Reply:
This is a good catch @ketankelkar - I am sorry if you mentioned a few weeks ago, not sure I was part of that discussion but if you run zos_lineinfile multiple times and it comes back with `results['changed'] == true and yet its a failure, either we have a bug or there is on in dmod.
We can investigate that in a separate git issue but that needs to exist, did you create an issue for it, put details and recreate steps into it? 

There is only one place changed is set to true and that is [here](https://github.com/ansible-collections/ibm_zos_core/blob/ad0506cd66563c36c3c4193f373842f7b16eb474/plugins/modules/zos_lineinfile.py#L507), so that means `dmod` failed and either we did not inspect correctly or they returned changed = true. 

This needs to be followed through.

Reply:
I guess it could go along with another little task for nexts iterations

Reply:
Is kind of solve with the last push

Reply:
Ok, after speaking with you , you were able to resolve this in the test code, so it was not no an issue in the module ore dmod code is how I understood it. I will resolve this.